agent/mibgroup/if-mib/data_access/interface.c

```diff
@@ -857,3 +899,61 @@ _free_interface_config(void)
     }
     conf_list = NULL;
 }
+
+/**---------------------------------------------------------------------*/
+/*
+ * include interface config token
+ */
+/**
+ */
+static void
+_parse_include_if_config(const char *token, char *cptr)
+{
+    netsnmp_include_if_list *if_ptr, *if_new;
+    char                   *name, *st;
+
+    name = strtok_r(cptr, " \t", &st);
+    if (!name) {
+        config_perror("Missing NAME parameter");
+        return;
+    }
+    if_ptr = include_list;
+    while (if_ptr)
+        if (strcmp(if_ptr->name, name))
+            if_ptr = if_ptr->next;
+        else
+            break;
+    if (if_ptr) {
+        config_pwarn("Duplicate include interface specification");
+        return;
+    }
```

Comment:
So this loop only checks whether the first interface name is unique? Shouldn't this check be moved inside the while(name) loop?

Reply:
You're right.  I just took Collet original patch without checking this.  I'll test a fix for this.

Reply:
So when I change this to a single loop, the IF-MIB interface filtering (index_list) doesn't
get initialized until after init_mib_modules() so that for the first 5 minutes or so after
snmpd is started, I see all the interfaces (over 1000).  After about 5 minutes, my interface
list is down to about 54 (which I configured with "include_ifmib_iface_prefix". 

Do you know of a way to get the "include" list to take effect right when snmpd is started?
It looks like the IF-MIB initialization (with a timer) is started before the new configuration
is parsed and my index_list is created.  I think this is caused this.

```
+    /* check for duplicate prefix configuration */
+    while (name != NULL) {
+        for (if_ptr = include_list; if_ptr; if_ptr = if_ptr->next) {
+            if (strncmp(name, if_ptr->name, strlen(if_ptr->name)) == 0) {
+                config_pwarn("Duplicate include interface prefix specification");
+                return;
+            }
+        }
+        /* now save the prefixes */
+        if_new = SNMP_MALLOC_TYPEDEF(netsnmp_include_if_list);
+        if (!if_new) {
+            config_perror("Out of memory");
+            return;
+        }
+        if_new->name = strdup(name);
+        if (!if_new->name) {
+            config_perror("Out of memory");
+            free(if_new);
+            return;
+        }
+        if_new->next = include_list;
+        include_list = if_new;
+        name = strtok_r(NULL, " \t", &st);
+    }
+}

```

Reply:
How about moving snmpd_register_config_handler("include_ifmib_iface_prefix", ...) from init_interface() into init_mib_modules(), just before the netsnmp_access_interface_init() call?

Reply:
I tried moving the call to snmpd_register_config_handler for the "include_ifmib_iface_prefix"
to the beginning of init_mib_modules:

```
@@ -65,6 +65,8 @@ init_mib_modules(void)
 {
     static int once = 0;
 
+    netsnmp_include_interface_init();
+
```
It still takes 5 minutes for my interface count to drop from 1056 to 56.
I would venture to guess that the initialization is happening without
my check for index_list of prefixes.  I'm probably just missing a call
for the check: 

```
+ if (!netsnmp_access_interface_include(ifrp->ifr_name)) 
+            continue; 
+ 

```